markov/tmatrix.R

```diff
@@ -1079,17 +1081,36 @@ get_results <- function(bat, pitch, stad, home, temp, date,
   #initialize empty lists/vectors
   candidate_batters <- list()
   candidate_pitchers <- list()
-  last_names <- vector()
+  id <- vector()
   
-  for (batter in bat)
-  {
-    candidate_batters <- append(candidate_batters, sing$GetPlayers(name=batter))
-    last_names <- append(last_names, word(batter, -1))
+  for (batter in bat) {
+    if (is.numeric(batter)) {
+      candidate_batters <- append(candidate_batters, sing$GetPlayers(id=batter)[[1]])
+      id <- append(id, batter)
+    }
+    else {
```

Comment:
Do you still need the non-numeric else code?

Reply:
I have it here for people who want to use the command line - if you want to avoid "Will Smith" confusion I can take out that portion

Reply:
I think you should remove it and just choose the first player with that name (possibly with a warning that you randomly picked the first player).